### ISSUE TITLE
Unable to create oplogs (error in creating a new API key)

### ISSUE THREAD

**Issue by 0x413x4** [GhostManager/Ghostwriter#197]
## Issue

Hiya,

I am experiencing some issues with the latest release. Essentially, I am unable to create Oplogs. Any attempts results in the following error messages:

```
TransactionManagementError at /oplog/create/
An error occurred in the current transaction. You can't execute queries until the end of the 'atomic' block.
```

and

```
Could not generate an API key for your new operation log – contact your admin!
```

Looking a bit further into it, I tracked it down to when Ghostwriter attempts to create a new API key for the oplog. If I try to create a new API key through the Admin Panel, I get the following error message:

```
DataError at /admin/rest_framework_api_key/apikey/add/
value too long for type character varying(100)
```

It looks like this may be an issue with the `djangorestframework-api-key` package. They recently pushed a commit to fix a similar issue 4 days ago (as of this writing): 

- https://github.com/florimondmanca/djangorestframework-api-key/issues/192

Have you experienced any issues on your end?

## Reproduction

Using a fresh install of Ghostwriter - tested with master (as of the 6th of March 2022) branch and v2.2.3 release:

- go to the Admin Panel
- Under API KEY PERMISSIONS, click **API Keys**
- click on **ADD API KEY**
- enter a name for the key (`my-new-api-key` was selected for this example)
- click on **Save**

## Expected behaviour

A new API key is created.

## Actual behaviour

The following error message is observed:

```
DataError at /admin/rest_framework_api_key/apikey/add/
value too long for type character varying(100)
```

**Comment by chrismaddalena** [comment]
Hey @0x413x4 , thanks for bringing this to my attention and providing such detail. This PR merged the code to fix this issue with argon-cffi:

https://github.com/florimondmanca/djangorestframework-api-key/pull/193

It's been inactive since January, but discussions resumed a few days ago. The bad news is the maintainer commented that they don't have the time to work on the project, but the contributor of the PR has offered to create a PR for a new release. Once a new release is out, I can update the requirements to use the latest djangorestframework-api-key.

We're moving away from Django REST Framework to remove it as a dependency, so we should be able to avoid issues like this in the future.

In the meantime, the PR that sets `max_length` to 150 for the hashed keys is in the main branch, so you should be able to install it using `pip` to pull directly from GitHub. The Django container does not have `git` installed by default, so you will need to install it.

I have tried this and have an issue getting `pip` to not just install the latest release. A command like this should work to install the package based on the (as of right now) latest commit from that PR:

`docker-compose -f local.yml run --rm django apk add git && pip install git+https://github.com/florimondmanca/djangorestframework-api-key.git@c8f8a8994bf1881b1aadbcdac06de0bd444ccf25#egg=djangorestframework-api-key`

This assumes you are in the Ghostwriter folder on the server. Replace `local.yml` with `production.yml` depending on the environment.

Whenever the release is ready I will update _base.py_ with the new release version.

**Comment by chrismaddalena** [comment]
I just realized my mistake. First, you'll need to use `exec`, not `run`, for the above command. However, I wasn't having luck getting this replacement installation working.

`docker-compose -f local.yml exec django apk add git && pip uninstall -y djangorestframework-api-key && pip install -e git+https://github.com/florimondmanca/djangorestframework-api-key@c8f8a8994bf1881b1aadbcdac06de0bd444ccf25#egg=djangorestframework-api-key`

The package installs, but API keys will still fail to generate. I think the model migrations aren't running during this process.

What does work is:

1. Open _base.py_ and replace line 31 with: `-e git+https://github.com/florimondmanca/djangorestframework-api-key@master#egg=djangorestframework-api-key`
2. Open _compose/<ENV>/django/Dockerfile_ and edit lines 24 and 25 like below (replace `<ENV>` with local or podcution base don your environment)
3. Run `docker-compose -f local.yml build`

```
    && pip install --no-cache-dir -U setuptools pip \
    && apk --no-cache add git
```

During the build, the Dockerfile will now install `git` and install the DRF API Key package from the GitHub repo.

I had the time to do so, so I have submitted a PR for a new release of `djangorestframework-api-key`. We'll be in good shape if that gets merged and a new release is tagged.
